lib/payonline/payment_gateway.rb

```diff
@@ -46,13 +16,29 @@ def filter_params
     }
 
     def initialize(params = {})
-      @params = Params.new(params.with_indifferent_access)
+      @params = prepare_params(params)
     end
 
     def payment_url(type: :card, language: :ru)
       params = Payonline::Signature.sign_params(@params, SIGNED_PARAMS)
 
       "#{BASE_URL}/#{language}/payment/#{PAYMENT_TYPE_URL[type]}?#{params.to_query}"
     end
+
+    private
+
+    def prepare_params(params)
+      params.with_indifferent_access
+        .slice(*REQUIRED_PARAMS)
```

Comment:
I use this style everywhere. sorry.
In my opinion readability will not change if your dot stays on the first line. But you forgot about debugging.